### DISCUSSION TITLE
[RAG] Hybrid search does not seem to work with Chinese/Japanese

### DISCUSSION THREAD

**Discussion by zhangchn** [open-webui/open-webui#6721] [Issues]
The problem with Chinese/Japanese is there is no spaces as a word boundary in these languages. The BM25Retriever from langchain use `split` as default preprocess function, and `split` only deals with languages with spaces between words.

To handle Chinese/Japanese properly, a competent tokenizer, like Jieba or Janome shall be assigned as parameter in `BM25Retriever.from_texts` call.

**Comment by zhangchn** [comment]
FYI: https://en.wikipedia.org/wiki/Category:Writing_systems_without_word_boundaries

**Comment by zhangchn** [comment]
See also: https://github.com/run-llama/llama_index/issues/13866

**Comment by aomiks** [comment]
老兄，请教个问题，我通过pip install open-webui -i https://pypi.tuna.tsinghua.edu.cn/simple安装的，windows11系统，在知识库上传文档报错：400: 'NoneType' object is not iterable，我试着用空txt一点点加文字，发现1.4K上传没问题，到了2K好像就不行了，内容都是同一段文字复制粘贴的，管理员面板设置的上传文档大小和数量都是默认留空无限制，请问这种情况你们没有吗？估计会是哪里的问题呢？
![image](https://github.com/user-attachments/assets/faa6b1d5-f733-4ddd-bb1b-82e4fb36fe6e)
